flickit-assessment-kit/src/test/java/org/flickit/assessment/kit/application/service/assessmentkit/update/impl/QuestionUpdateKitPersisterTest.java

```diff
@@ -142,49 +121,95 @@ void testQuestionUpdateKitPersister_SameInputsAsDatabaseData_NoChange() {
-        var attribute = createAttribute(ATTRIBUTE_CODE1, ATTRIBUTE_TITLE1, 1, "", 1);
-        var savedImpact = createQuestionImpact(attribute.getId(), levelTwo.getId(), 1, savedQuestion.getId());
-        var answerOption1 = createAnswerOption(savedQuestion.getAnswerRangeId(), OPTION_TITLE, OPTION_INDEX1);
-        var answerOption2 = createAnswerOption(savedQuestion.getAnswerRangeId(), OPTION_TITLE, OPTION_INDEX2);
-        var savedOptionImpact1 = createAnswerOptionImpact(answerOption1.getId(), 0);
-        var savedOptionImpact2 = createAnswerOptionImpact(answerOption2.getId(), 1);
-        savedImpact.setOptionImpacts(List.of(savedOptionImpact1, savedOptionImpact2));
-        savedQuestion.setOptions(List.of(answerOption1, answerOption2));
-        savedQuestion.setImpacts(List.of(savedImpact));
-        savedQuestionnaire2.setQuestions(List.of(savedQuestion));
-
-        when(loadAnswerOptionsByQuestionPort.loadByQuestionId(any(), eq(savedKit.getActiveVersionId()))).thenReturn(List.of(answerOption1, answerOption2));
+        KitContext kitContext = createKitContext();
+        kitContext.questionnaire().setQuestions(List.of(kitContext.question()));
 
-        var dslMaturityLevelTwo = MaturityLevelDslModelMother.domainToDslModel(levelTwo());
-        var dslQuestionnaire = QuestionnaireDslModelMother.domainToDslModel(savedQuestionnaire2);
-        var dslAnswerOption1 = answerOptionDslModel(1, OPTION_TITLE, OPTION_VALUE1);
-        var dslAnswerOption2 = answerOptionDslModel(2, OPTION_TITLE, OPTION_VALUE2);
-        List<AnswerOptionDslModel> dslAnswerOptionList = List.of(dslAnswerOption1, dslAnswerOption2);
-        Map<Integer, Double> optionsIndexToValueMap = new HashMap<>();
-        optionsIndexToValueMap.put(dslAnswerOption1.getIndex(), 0D);
-        optionsIndexToValueMap.put(dslAnswerOption2.getIndex(), 1D);
-        var dslImpact = QuestionImpactDslModelMother.questionImpactDslModel(ATTRIBUTE_CODE1, dslMaturityLevelTwo, null, optionsIndexToValueMap, 1);
-        var dslQuestion = QuestionDslModelMother.questionDslModel(QUESTION_CODE1, 1, QUESTION_TITLE1, null, "c-" + QUESTIONNAIRE_TITLE1, List.of(dslImpact), dslAnswerOptionList, Boolean.FALSE, Boolean.TRUE);
-        AssessmentKitDslModel dslKit = AssessmentKitDslModel.builder()
-            .questionnaires(List.of(dslQuestionnaire))
-            .questions(List.of(dslQuestion))
-            .build();
+        var expectedAnswerRangeId = kitContext.answerOption1.getAnswerRangeId();
+        var expectedQuestionId = 251L;
+        var expectedQuestionImpactId = 56115L;
+        when(createAnswerRangePort.persist(any())).thenReturn(expectedAnswerRangeId);
+        when(createQuestionPort.persist(any())).thenReturn(expectedQuestionId);
+        when(loadAnswerOptionsByQuestionPort.loadByQuestionId(any(), eq(savedKit.getActiveVersionId()))).thenReturn(List.of(kitContext.answerOption1(), kitContext.answerOption2()));
```

Comment:
Can be change to  
`when(loadAnswerOptionsByQuestionPort.loadByQuestionId(expectedQuestionId, savedKit.getActiveVersionId())).thenReturn(List.of(kitContext.answerOption1(), kitContext.answerOption2()));`